datalad_next/create_sibling_webdav.py

```diff
@@ -343,16 +344,22 @@ def _dummy(ds, refds, **kwargs):
 
         # Generate a sibling for dataset "ds", and for sub-datasets if recursive
         # is True.
-        for res in ds.foreach_dataset(_dummy,
-                                      return_type='generator',
-                                      result_renderer='disabled',
-                                      recursive=recursive,
-                                      recursion_limit=recursion_limit):
-            # unwind result generator
-            for partial_result in res.get('result', []):
+        if not recursive:
```

Comment:
This special case could be implemented in a cheaper fashion using `recursion_limit=0` if not recursive.

```sh
(datalad-dev) mih@meiner /tmp/rec (git)-[master] % dl foreach-dataset -R 0 ':'
foreach-dataset(ok): /tmp/rec (dataset)
(datalad-dev) mih@meiner /tmp/rec (git)-[master] % dl foreach-dataset ':'
foreach-dataset(ok): /tmp/rec (dataset)
foreach-dataset(ok): /tmp/rec/s1 (dataset)
foreach-dataset(ok): /tmp/rec/s2 (dataset)
```

Reply:
Agree. Will do.